### PR TITLE
Packages: Extract dom package and make it maintained with Lerna

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Related issues: #3955, #6594.

> This PR tries to provide Lerna setup similar to what we have in `WordPress/packages`. This would allow publishing source code of individual modules to npm to make them available for all plugin developers.

This PR extract new `dom` package which contains all DOM utils.

## Open question

Question is if we should also provide `deprecation` message for all moved functions from `@wordpress/utils`?

## How has this been tested?
Manually:
* Removed `node_modules` folder.
* `npm install`
* `npm test`
* `npm run dev`
* `npm run build`
* `npm run package-plugin`

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -62,6 +62,10 @@ module.exports = {
 				selector: 'ImportDeclaration[source.value=/^data$/]',
 				message: 'Use @wordpress/data as import path instead.',
 			},
+			{
+				selector: 'ImportDeclaration[source.value=/^dom$/]',
+				message: 'Use @wordpress/dom as import path instead.',
+			},
```
Comment (ntwb) [inline .eslintrc.js]: 👍

**Review by ntwb — APPROVED** [review]
LGTM 👍

**Comment by ntwb** [comment]
Possible merge conflict with #6756 when it comes time to merge

**Comment by gziolo** [comment]
Added deprecation for all dom utils in `wp.utils` and rebased with `master`.

**Comment by gziolo** [comment]
There are packages which depend on `react` and `react-dom` and it produces warnings when they are only hoisted, too early to remove them from root file…

The same applies to `moment` and `tinymce`, the only difference is that nothing warns :)
Removing c367c49 from this PR.

**Pull request merged**